src/composables/auth.ts

```diff
@@ -1,50 +1,91 @@
 import { BiometricAuth } from '@aparajita/capacitor-biometric-auth';
 import {
+  computed,
   readonly,
   ref,
   watch,
 } from 'vue';
+import { generateMnemonic, mnemonicToSeed, validateMnemonic } from '@aeternity/bip39';
 
 import { tg as t } from '@/popup/plugins/i18n';
-import { IS_MOBILE_APP, MODAL_ENABLE_BIOMETRIC_LOGIN, MODAL_BIOMETRIC_LOGIN } from '@/constants';
+import {
+  IS_MOBILE_APP,
+  MODAL_ENABLE_BIOMETRIC_LOGIN,
+  MODAL_BIOMETRIC_LOGIN,
+  IS_IOS,
+  STORAGE_KEYS,
+  IS_EXTENSION,
+  MODAL_PASSWORD_LOGIN,
+  IS_OFFSCREEN_TAB,
+} from '@/constants';
 import {
   authenticateWithPassword,
+  encrypt,
+  endSession,
+  generateEncryptionKey,
   getSessionEncryptionKey,
   sleep,
+  startSession,
   watchUntilTruthy,
 } from '@/utils';
+
+import { WalletStorage } from '@/lib/WalletStorage';
+import { SecureMobileStorage } from '@/lib/SecureMobileStorage';
+
+import migrateMnemonicVuexToComposable from '@/migrations/002-mnemonic-vuex-to-composable';
+import migrateMnemonicCordovaToIonic from '@/migrations/008-mnemonic-cordova-to-ionic';
+
 import { useUi } from './ui';
 import { useModals } from './modals';
-import { useAccounts } from './accounts';
 import { createCustomScopedComposable } from './composablesHelpers';
+import { useSecureStorageRef } from './secureStorageRef';
+import { useAeSdk } from './aeSdk';
 
-export const useAuth = createCustomScopedComposable(() => {
-  let composableInitialized = false;
-
-  const biometricAuth = {
-    available: false,
-    updating: false,
-    checked: false,
-  };
-
-  /** Common state for both biometric or password protection */
-  const isAuthenticated = ref(false);
-  const isAuthenticating = ref(false);
+const CHECK_FOR_SESSION_KEY_INTERVAL = 5000;
 
+export const useAuth = createCustomScopedComposable(() => {
   const {
     isBiometricLoginEnabled,
     lastTimeAppWasActive,
     secureLoginTimeout,
     isAppActive,
     setBiometricLoginEnabled,
+    setLoaderVisible,
   } = useUi();
   const { openModal } = useModals();
-  const {
-    isLoggedIn,
-    encryptionKey,
-    setEncryptionKey,
-    openPasswordLoginModal,
-  } = useAccounts();
+
+  /** Common state for both biometric or password protection */
+  const isAuthenticated = ref(false);
+  const isAuthenticating = ref(false);
+  const isMnemonicRestored = ref(false);
+  const encryptionKey = ref<CryptoKey | null>();
+
+  const [mnemonicEncrypted, mnemonic] = useSecureStorageRef<string>(
```

Comment:
This is important @CedrikNikita @martinkaintas : the mnemonic was moved from `accounts` composable to `auth`. Are OK with this?